src/hooks/airdrop/stateHelpers/useAirdropPolling.ts

```diff
@@ -0,0 +1,60 @@
+import { useAirdropStore } from '@app/store';
+import {
+    fetchAllUserData,
+    fetchLatestXSpaceEvent,
+    fetchPollingFeatureFlag,
+} from '@app/store/actions/airdropStoreActions';
+import { useCallback, useEffect, useRef } from 'react';
+
+const DEBOUNCE_DELAY = 1000; // 1 second delay
+
+export const useAirdropPolling = () => {
+    const pollingEnabled = useAirdropStore((s) => s.pollingEnabled);
+    const timeoutRef = useRef<NodeJS.Timeout>();
+
+    const fetchAirdropDataDebounced = useCallback(() => {
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+        timeoutRef.current = setTimeout(async () => {
+            await fetchAllUserData();
```

Comment:
sorry if i missed the context, i haven't caught up on the ws rework stuff properly 😅 

why are we getting all user data with polling now? i'm testing locally and the `user_id` ws events are still coming through (even while minimized), so is the the x space event 

was there an issue with user data or something? 😬 

is this just meant to be a fallback in case ws disconnects or ?

Reply:
maybe we can only poll if the socket isn't connected? i'm not sure why we have both? 😅 

also if it's just a fallback every minute seems like a lot if ws is still emitting every 15s while mining?